core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -573,7 +584,7 @@ private[kafka] class Acceptor(val endPoint: EndPoint,
   private def startProcessors(processors: Seq[Processor], processorThreadPrefix: String): Unit = synchronized {
     processors.foreach { processor =>
       KafkaThread.nonDaemon(
-        s"${processorThreadPrefix}-kafka-network-thread-$brokerId-${endPoint.listenerName}-${endPoint.securityProtocol}-${processor.id}",
+        s"${processorThreadPrefix}-kafka-network-thread-$nodeId-${endPoint.listenerName}-${endPoint.securityProtocol}-${processor.id}",
```

Comment:
When in combined mode (or in a JUnit test) we need a way to distinguish the processor for controller node 0 from the process for broker node 0.  Can we just have a boolean we pass to indicate "this is a socket server for a controller"?

That would also avoid the need to pass in an optional logcontext, too.